lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/task/CommunicationGroupClientImpl.java

```diff
@@ -191,6 +248,8 @@ public void initialize() {
         throw new RuntimeException("InterruptedException while waiting for initialization", e);
       }
 
+      updateTopology();
```

Comment:
updateTopology? IIRC this is to be called only by master task from user code. I think you are trying to use this to bootstrap active tasks. I am not sure if this is the right way. initialize() will be called by all the tasks.